### PR TITLE
88 categorize balls for pickup strategies

### PR DESCRIPTION
**Pull request by Greve2001**

### CODE REVIEW

File: remote/src/main/java/vision/BallPickupStrategy.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package vision;
+
+public enum BallPickupStrategy {
+    FREE, BORDER, CORNER
```
Comment (yech199) [inline remote/src/main/java/vision/BallPickupStrategy.java:4]: Need one for CROSS?

File: remote/src/main/java/vision/DetectionController.java [hunk 2]
```diff
@@ -176,6 +182,58 @@ private void detectCourse(VideoCapture capture) {
         HighGui.waitKey(this.refreshRate);
     }
 
+    /**
+     * Changes the BallPickupStrategy for each of the balls in the list, depending on how close they are to the borders.
```
Comment (yech199) [inline remote/src/main/java/vision/DetectionController.java:186]: Add TODO comment for cross and corners?

File: remote/src/main/java/vision/DetectionController.java [hunk 3]
```diff
@@ -176,6 +182,58 @@ private void detectCourse(VideoCapture capture) {
...
+     * @param balls The list of balls to be changed
+     * @param corners the corner of the box surrounding the balls.
+     */
+    private void categorizeBallsPickupStrategy(List<Ball> balls, Point[] corners) {
+        final double centimeterMargin = 5;
```
Comment (yech199) [inline remote/src/main/java/vision/DetectionController.java:191]: Width of the border?

**Review by yech199 — COMMENTED** [review]
Looks good, requested some small changes and comments :smile_cat:

**Comment by Greve2001** [comment]
I have added the cross into the categorization now

**Review by yech199 — APPROVED** [review]

**Pull request merged**